[Bug]: Pre-release daily workflows failing with zarr 3.0.0a0

### What happened?

All pre-release daily workflows are failing. https://github.com/hdmf-dev/hdmf-zarr/actions/runs/9559407286

It looks like they install the latest zarr 3.0.0 pre-release, 3.0.0a0, which was released last week.

I think this can be addressed by changing `from zarr.hierarchy import Group` to `from zarr import Group`. I will give it a try.

### Steps to Reproduce

```shell
Run tests with zarr 3.0.0a0
```


### Traceback

```shell
/opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/unit/test_io_convert.py:43: in <module>
    from hdmf_zarr.backend import (ZarrIO,
.tox/py312-prerelease/lib/python3.12/site-packages/hdmf_zarr/__init__.py:1: in <module>
    from .backend import ZarrIO
.tox/py312-prerelease/lib/python3.12/site-packages/hdmf_zarr/backend.py:11: in <module>
    from zarr.hierarchy import Group
E   ModuleNotFoundError: No module named 'zarr.hierarchy'
```


### Operating System

Linux

### Python Executable

Conda

### Python Version

3.12

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/hdmf-dev/hdmf-zarr/blob/dev/.github/CODE_OF_CONDUCT.md)
- [X] Have you checked the [Contributing](https://github.com/hdmf-dev/hdmf-zarr/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this bug was not already [reported](https://github.com/hdmf-dev/hdmf-zarr/issues)?

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,7 +30,7 @@ classifiers = [
 ]
 dependencies = [
     'hdmf>=3.9.0',
-    'zarr>=2.11.0',
+    'zarr>=2.11.0, <3.0', # pin below 3.0 until HDMF-zarr supports zarr 3.0
     'numpy>=1.24, <2.0', # pin below 2.0 until HDMF supports numpy 2.0
     'numcodecs>=0.9.1',
     'pynwb>=2.5.0',

